src/timesync/timesyncd-manager.c

```diff
@@ -383,10 +383,10 @@ static int manager_adjust_clock(Manager *m, double offset, int leap_sec) {
                   "  offset       : %+.3f sec\n"
                   "  freq offset  : %+li (%i ppm)\n",
                   tmx.status, tmx.status & STA_UNSYNC ? "unsync" : "sync",
-                  tmx.time.tv_sec, tmx.time.tv_usec / NSEC_PER_MSEC,
-                  tmx.constant,
+                  (long int)tmx.time.tv_sec, tmx.time.tv_usec / NSEC_PER_MSEC,
+                  (long int)tmx.constant,
                   (double)tmx.offset / NSEC_PER_SEC,
-                  tmx.freq, m->drift_ppm);
+                  (long int)tmx.freq, m->drift_ppm);
```

Comment:
```long int``` is 32 bits on ```i386``` and 64 bits on ```x86_64```. The reduction happens on ```x32``` only where ```tmx.freq``` is 64 bits but ```long int``` is 32 bits.

Please note that we are using ```%li``` as the format specifier which is the proper format specifier for ```long int``` after all.